artemis/db.py

```diff
@@ -376,6 +409,67 @@ def _get_task_deduplication_data(self, task: Task) -> str:
             }
         ).replace("\x00", " ")
 
+    def take_single_report_generation_task(self) -> Optional[ReportGenerationTask]:
+        with self.session() as session:
+            query = session.query(ReportGenerationTask).filter(
+                ReportGenerationTask.status == ReportGenerationTaskStatus.PENDING.value
+            )
+            if query.count() > 0:
+                return query[0]  # type: ignore
+        return None
+
+    def mark_report_generation_task_as_completed(
+        self,
+        task: ReportGenerationTask,
+        status: ReportGenerationTaskStatus,
```

Comment:
if task status is known from function name, should we not just have `ReportGenerationTaskStatus.done` in function directly? I mean, I dont see reason to pass it here

Reply:
changed function name to better reflect its purpose